### ISSUE TITLE
error: implicit declaration of function ‘asprintf’

### ISSUE THREAD

**Issue by yagop** [willemt/yabtorrent#14]
```
➜  /tmp  git clone https://github.com/willemt/YABTorrent
Cloning into 'YABTorrent'...
remote: Counting objects: 2222, done.
remote: Total 2222 (delta 0), reused 0 (delta 0), pack-reused 2222
Receiving objects: 100% (2222/2222), 1.10 MiB | 571.00 KiB/s, done.
Resolving deltas: 100% (1586/1586), done.
Checking connectivity... done.
➜  /tmp  cd YABTorrent 
➜  YABTorrent git:(master) make
python waf configure
Setting top to                           : /tmp/YABTorrent 
Setting out to                           : /tmp/YABTorrent/build 
Checking for 'gcc' (C compiler)          : /usr/bin/gcc 
Checking for library uv                  : yes 
'configure' finished successfully (0.044s)
python waf build
Waf: Entering directory `/tmp/YABTorrent/build'
[  1/237] c: src/bt_blacklist.c -> build/src/bt_blacklist.c.1.o
[  2/237] c: src/bt_choker_leecher.c -> build/src/bt_choker_leecher.c.1.o
[  3/237] c: src/bt_choker_seeder.c -> build/src/bt_choker_seeder.c.1.o
[  4/237] c: src/bt_blockrw_cache.c -> build/src/bt_blockrw_cache.c.1.o
[  5/237] c: src/bt_blockrw_mem.c -> build/src/bt_blockrw_mem.c.1.o
[  6/237] c: src/bt_download_manager.c -> build/src/bt_download_manager.c.1.o
[  7/237] c: src/bt_peer_manager.c -> build/src/bt_peer_manager.c.1.o
[  8/237] c: src/bt_piece.c -> build/src/bt_piece.c.1.o
[  9/237] c: src/bt_piece_db.c -> build/src/bt_piece_db.c.1.o
../src/bt_peer_manager.c: In function ‘bt_peermanager_add_peer’:
../src/bt_peer_manager.c:135:9: error: implicit declaration of function ‘asprintf’ [-Werror=implicit-function-declaration]
         asprintf(&peer->peer_id, "00000000000000000000");
         ^
cc1: all warnings being treated as errors

Waf: Leaving directory `/tmp/YABTorrent/build'
Build failed
 -> task in 'yabbt' failed (exit status 1): 
	{task 139765216812240: c bt_peer_manager.c -> bt_peer_manager.c.1.o}
['/usr/bin/gcc', '-Wno-error', '-Werror', '-Werror=format', '-Werror=int-to-pointer-cast', '-g', '-DLINUX', '-Werror=unused-variable', '-Werror=return-type', '-Werror=uninitialized', '-Werror=pointer-to-int-cast', '-Wcast-align', '-fPIC', '-I/tmp/YABTorrent/build/include', '-I/tmp/YABTorrent/include', '-I/tmp/YABTorrent/build/deps/config-re', '-I/tmp/YABTorrent/deps/config-re', '-I/tmp/YABTorrent/build/deps/linked-list-queue', '-I/tmp/YABTorrent/deps/linked-list-queue', '-I/tmp/YABTorrent/build/deps/fe', '-I/tmp/YABTorrent/deps/fe', '-I/tmp/YABTorrent/build/deps/bag', '-I/tmp/YABTorrent/deps/bag', '-I/tmp/YABTorrent/build/deps/pwp', '-I/tmp/YABTorrent/deps/pwp', '-I/tmp/YABTorrent/build/deps/bitfield', '-I/tmp/YABTorrent/deps/bitfield', '-I/tmp/YABTorrent/build/deps/event-timer', '-I/tmp/YABTorrent/deps/event-timer', '-I/tmp/YABTorrent/build/deps/linked-list-hashmap', '-I/tmp/YABTorrent/deps/linked-list-hashmap', '-I/tmp/YABTorrent/build/deps/strndup', '-I/tmp/YABTorrent/deps/strndup', '-I/tmp/YABTorrent/build/deps/meanqueue', '-I/tmp/YABTorrent/deps/meanqueue', '-I/tmp/YABTorrent/build/deps/heap', '-I/tmp/YABTorrent/deps/heap', '-I/tmp/YABTorrent/build/deps/pseudolru', '-I/tmp/YABTorrent/deps/pseudolru', '-I/tmp/YABTorrent/build/deps/chunkybar', '-I/tmp/YABTorrent/deps/chunkybar', '-I/tmp/YABTorrent/build/deps/sha1', '-I/tmp/YABTorrent/deps/sha1', '-I/tmp/YABTorrent/build/deps/bitstream', '-I/tmp/YABTorrent/deps/bitstream', '-I/tmp/YABTorrent/build/deps/array-avl-tree', '-I/tmp/YABTorrent/deps/array-avl-tree', '../src/bt_peer_manager.c', '-c', '-o', '/tmp/YABTorrent/build/src/bt_peer_manager.c.1.o']
Makefile:4: recipe for target 'lets_build_this' failed
make: *** [lets_build_this] Error 1
➜  YABTorrent git:(master) ✗ 
```

**Comment by willemt** [comment]
Thanks for the issue report!

This should now be fixed by https://github.com/willemt/yabtorrent/commit/17bd53302e2c6e52da9c3fdfe350af780b93ed43

Please let me know if you find any other issues!

**Comment by yagop** [comment]
Thanks a lot! Working for src, tests were failing too I PR a fix for them #15
